tests/Doctrine/Common/Lexer/TokenTest.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\Common\Lexer;
+
+use Doctrine\Common\Lexer\Token;
+use Doctrine\Deprecations\PHPUnit\VerifyDeprecations;
+use PHPUnit\Framework\TestCase;
+
+final class TokenTest extends TestCase
+{
+    use VerifyDeprecations;
+
+    public function testIsA(): void
+    {
+        $token = new Token('foo', 'string', 1);
+
+        self::assertTrue($token->isA('string'));
+        self::assertTrue($token->isA('int', 'string'));
+        self::assertFalse($token->isA('int'));
+    }
+
+    public function testArrayAccessIsDeprecated(): void
+    {
+        $token = new Token('foo', 'string', 1);
+        self::expectDeprecationWithIdentifier('https://github.com/doctrine/lexer/pull/79');
+        self::assertSame('foo', $token['value']);
+    }
```

Comment:
If we really keep ArrayAccess, then we should add tests for the other methods too.

Reply:
Thanks! You helped me spot a flaw in the implementation! unsetting a property removes it, the right way seems to be using `= null` instead.